03_Fiches_thematiques/Fiche_import_fichiers_parquet.qmd

```diff
@@ -0,0 +1,253 @@
+
+::: {.callout-recommandation .icon}
+
+- Pour les **non statisticiens**, il est recommandé d'utiliser **le format csv** ;
+- Pour les **statisticiens**, il est recommandé d'utiliser **le format Parquet**.
+
+:::
+
+La suite de cette fiche s'adresse donc aux statisticiens.
+
+## Caractéristiques de Parquet 
+
+**Parquet présente quelques propriétés qui le distinge des formats de fichiers plus populaires :**
+
+- Parquet repose sur un **stockage en mode colonne**. Ainsi seront stockées dans un premier temps toutes les données du premier attribut, puis seulement dans un second temps les données du deuxième attribut et ainsi de suite...  
+Dans un contexte analytique, cette organisation des données génère plusieurs avantages dont les principaux sont : 
+- **Un gain de rapidité lors de la lecture des données** ;
+- **La possibilité d'avoir un haut niveau de compression**.
+
+**Un fichier Parquet contient à la fois les données et les métadonnées**. Ces métadonnées écrites en bas de page d'un fichier enregistrent le schéma de ce fichier selon 3 niveaux : fichier, bloc et en-tête de page (voir [ici](https://parquet.apache.org/docs/file-format/metadata/) pour en savoir plus). Ce sont ces métadonnées qui font en sorte que la lecture des données Parquet soit optimisée et sans risque d’altération.
+
+Enfin, Parquet dispose d'un autre avantage important : **il sait modéliser des types complexes**. Il s'agit par exemple d'une colonne qui contient une structure hiérarchique ou du champ géométrique d'une couche SIG.
+
+Pour en savoir plus notamment sur la comparaison entre les formats Parquet et csv, consultez 
+[l'excellent chapitre sur le sujet](https://pythonds.linogaliana.fr/reads3/#le-format-parquet) écrit 
+par Lino Galiana dans son cours `Python pour la data-science`.  
+
+Grâce aux travaux d'Apache, **les fichiers aux format Parquet sont inter-opérables** c'est-à-dire qu'ils peuvent être lus par plusieurs langages informatiques : [C](https://arrow.apache.org/docs/c_glib/), [C++](https://arrow.apache.org/docs/cpp/), [C#](https://github.com/apache/arrow/blob/main/csharp/README.md), [Go](https://godoc.org/github.com/apache/arrow/go/arrow), [Java](https://arrow.apache.org/docs/java/), [JavaScript](https://arrow.apache.org/docs/js/), [Julia](https://arrow.juliadata.org/stable/), [MATLAB](https://github.com/apache/arrow/blob/main/matlab/README.md), [Python](https://arrow.apache.org/docs/python/), [Ruby](https://github.com/apache/arrow/blob/main/ruby/README.md), [Rust](https://docs.rs/crate/arrow/) et bien entendu [R](https://arrow.apache.org/docs/r/).
+
+S'il est très efficace pour l'analyse de données, **Parquet est en revanche peu adapté à l'ajout de données en continu ou à la modification fréquente de données existantes**.  
+Pour cette utilisation, le statisticien privilégiera un système de gestion de base de données comme par exemple [PostgreSQL](https://www.postgresql.org/).
+
+
+## Lire un fichier Parquet avec R
```

Comment:
Je pense qu'on pourrait partir du même cas que dans le post de blog `Polars` de @romaintailhurat : on écrit la BPE au format parquet et tous les exemples qui suivent s'appuient sur ce fichier

Reply:
C'est noté @linogaliana je vais adapter les exemples avec la BPE